src/common/components/build-status/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import { Link } from 'react-router';
+import moment from 'moment';
+
+import { BuildStatusConstants } from '../../helpers/snap-builds';
+
+import styles from './buildStatus.css';
+
+const BuildStatus = (props) => {
+
+  const {
+    link,
+    status,
+    statusMessage,
+    dateStarted
+  } = props;
+
+  const statusStyle = {
+    [BuildStatusConstants.SUCCESS]: styles.success,
+    [BuildStatusConstants.ERROR]: styles.error,
+    [BuildStatusConstants.PENDING]: styles.pending
+  };
+
+  let humanDateStarted;
+
+  if (dateStarted) {
```

Comment:
Build status and date handling mostly copied from current `BuildRow` component.
`BuildRow` need to be redesigned as MAAS table anyway, so it can now reuse `BuildStatus` as part of it.